### PR TITLE
Move location of eventhandler registration

### PR DESCRIPTION
**Pull request by sargun**
This moves the event handler registration until after the cache
is in-sync.

It makes it so we can use the log object from the context,
rather than having to use the global logger

The cache race condition of the cache starting while the reactor
is being added wont exist because we wait for the cache
to startup / go in sync prior to adding it.

### CODE REVIEW

**Comment by cpuguy83** [comment]
Weren't these recently moved out of Run?

**Comment by sargun** [comment]
Yeah, I didn't realize I could move it below sync. I just had to make it so it didn't run concurrently with start.

File: node/podcontroller.go [hunk 1]
```diff
@@ -161,14 +161,35 @@ func NewPodController(cfg PodControllerConfig) (*PodController, error) {
...
+// It will block until the context is cancelled, at which point it will shutdown the work queue and wait for workers to finish processing their current work items.
+func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) error {
+	defer pc.k8sQ.ShutDown()
+
+	podStatusQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "syncPodStatusFromProvider")
+	pc.runSyncFromProvider(ctx, podStatusQueue)
+	pc.runProviderSyncWorkers(ctx, podStatusQueue, podSyncWorkers)
+	defer podStatusQueue.ShutDown()
+
+	// Wait for the caches to be synced *before* starting workers.
```
Comment (cpuguy83) [inline node/podcontroller.go]: "Workers" are started, we just haven't added event handlers.

Comment (sargun) [reply node/podcontroller.go]: Let me rearrange this code to reflect the truth. I don’t actually see any downside of starting the workers for k8s->provider. The other way might be problematic though.

File: node/podcontroller.go [hunk 2]
```diff
@@ -161,14 +161,35 @@ func NewPodController(cfg PodControllerConfig) (*PodController, error) {
 		resourceManager: rm,
 		ready:           make(chan struct{}),
 		recorder:        cfg.EventRecorder,
-		k8sQ:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "syncPodsFromKubernetes"),
+
+		k8sQ: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "syncPodsFromKubernetes"),
```
Comment (gabrielhartmann) [inline node/podcontroller.go]: weird formatting

**Pull request merged**